app/services/store_policy_resolver.py

```
@@ -0,0 +1,641 @@
+            raise PolicyNotFound(f"Policy UUID {policy_uuid} not found (may have been deleted)")
+
+        # Parse and validate payload per SPEC-STORE-001
+        try:
+            return StorePolicyConfigParser.parse(
+                payload=store_product.payload,
+                policy_uuid=store_product.policy_uuid,
+                class_id=store_product.class_id,
+                created_at=store_product.created_at,
+            )
+        except (PolicyParseError, PolicyValidationError) as e:
+            raise StorePolicyError(f"Policy {policy_uuid} validation failed: {str(e)}")
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)